src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1820,20 +1820,32 @@ bool checkValid(MethodSymbol candidate, bool isCandidate, DiagnosticBag specific
                         diagnostics.Add(ErrorCode.ERR_NoMainInClass, mainType.Locations.First(), mainType);
                     }
                 }
-                else if (viableEntryPoints.Count > 1)
-                {
-                    viableEntryPoints.Sort(LexicalOrderSymbolComparer.Instance);
-                    var info = new CSDiagnosticInfo(
-                         ErrorCode.ERR_MultipleEntryPoints,
-                         args: Array.Empty<object>(),
-                         symbols: viableEntryPoints.OfType<Symbol>().AsImmutable(),
-                         additionalLocations: viableEntryPoints.Select(m => m.Locations.First()).OfType<Location>().AsImmutable());
-
-                    diagnostics.Add(new CSDiagnostic(info, viableEntryPoints.First().Locations.First()));
-                }
                 else
                 {
-                    entryPoint = viableEntryPoints[0];
+                    foreach (var viableEntryPoint in viableEntryPoints)
+                    {
+                        if (viableEntryPoint.UnmanagedCallersOnlyAttributeData is { } data)
+                        {
+                            Debug.Assert(!ReferenceEquals(data, UnmanagedCallersOnlyAttributeData.Uninitialized));
+                            diagnostics.Add(ErrorCode.ERR_EntryPointCannotBeUnmanagedCallersOnly, viableEntryPoint.Locations.First());
+                        }
+                    }
+
+                    if (viableEntryPoints.Count > 1)
```

Comment:
>if (viableEntryPoints.Count > 1) [](start = 20, length = 32)

Is this ever `false`?

Reply:
Yes, when the user has more than one valid entry point. I have several tests that show this.

---
In reply to: [482327460](https://github.com/dotnet/roslyn/pull/47311#discussion_r482327460) [](ancestors = 482327460)